File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06IO2/BridgePatern/BridgePatern/ShapeTriangle.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePatern
{
    class ShapeTriangle : Shape
    {
        private static Brush bg = Brushes.Green;
        private float ax, ay, bx, by, cx, cy;
        public ShapeTriangle(string text)
        {
            ParseParams(text, v => ax = v, v => ay = v, v => bx = v, v => by = v, v => cx = v, v => cy = v);
        }
        public override void Draw(Graphics g)
        {
            g.FillPolygon(bg, new PointF[] { new PointF(ax,ay), new PointF(bx, by), new PointF(cx, cy) });
        }
        public override string ToXml()
        {
            return $"<rectangle ax=\"{ax}\" ay=\"{ay}\" bx=\"{bx}\" by=\"{by}\" />";
        }
        public override string ToString()
        {
            return $"Rectangle(ax:{ax},ay:{ay},bx:{bx},by:{by})";
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022w/ID07IO1/Bridge/src/ShapeTriangle.java

import java.awt.Color;
import java.awt.Graphics;

public class ShapeTriangle extends Shape
{
   private float ax,ay,bx,by,cx,cy;
   public ShapeTriangle(String text)
   {
      float[] params=parseParams(text,6);
      ax=params[0];
      ay=params[1];
      bx=params[2];
      by=params[3];
      cx=params[4];
      cy=params[5];
   }

   @Override public void draw(Graphics g)
   {
      g.setColor(Color.blue);
      //g.drawPolyline
      g.fillPolygon
      (
         new int[] { Math.round(ax),Math.round(bx),Math.round(cx),Math.round(ax) },
         new int[] { Math.round(ay),Math.round(by),Math.round(cy),Math.round(ay) },
         4
      );
   }

   @Override public String toString()
   {
      return String.format("%g, %g; %g, %g; %g, %g",ax,ay,bx,by,cx,cy);
   }
}
